src/rgw/rgw_rest_s3.cc

```diff
@@ -2922,9 +2946,28 @@ void RGW_Auth_S3::init_impl(RGWRados* store)
   const string& ldap_searchdn = store->ctx()->_conf->rgw_ldap_searchdn;
   const string& ldap_dnattr =
     store->ctx()->_conf->rgw_ldap_dnattr;
+  string ldap_bindpw;
 
-  ldh = new rgw::LDAPHelper(ldap_uri, ldap_binddn, ldap_searchdn,
-			    ldap_dnattr);
+  if (store->ctx()->_conf->rgw_ldap_secret.empty()) {
+    ldout(store->ctx(), 10)
+      << __func__ << " LDAP auth no rgw_ldap_secret file found in conf"
+      << dendl;
+  } else {
+    char bindpw[1024];
+    memset(bindpw, 0, 1024);
+    int pwlen = safe_read_file("" /* base */,
+			       store->ctx()->_conf->rgw_ldap_secret.c_str(),
+			       bindpw, 1023);
+    if (pwlen) {
+      ldap_bindpw = bindpw;
+      boost::algorithm::trim(ldap_bindpw);
+      if (ldap_bindpw.back() == '\n')
+	ldap_bindpw.pop_back();
+    }
+  }
```

Comment:
looks like this block is duplicated in librgw.cc, consider moving into a helper function in rgw_ldap.h